api/base/serializers.py

```diff
@@ -172,6 +172,15 @@ def should_be_none(self, instance):
         return not isinstance(self.field, RelationshipField)
 
 
+class HideIfWikiDisabled(ConditionalField):
+    """
+    If wiki is disabled, don't show relationship field
+    """
+
+    def should_hide(self, instance):
+        return 'wiki' not in instance.get_addon_names()
```

Comment:
Please make a new version of the api (2.8?) and only have this happen if it's that version or higher.